### DISCUSSION TITLE
Listen to frontlight toggle event on Pocketbook

### DISCUSSION THREAD

**Discussion by dmalinovsky** [koreader/koreader#12524] [Q&A] [answered]
I have Pocketbook Inkpad Color 3 device. To toggle frontlight, I've mapped Menu button (one of 4 physical buttons, the right one) to do this action. It works in both stock reader and KOReader, so far so good.

I would like to update KOReader's status bar whenever the frontlight is toggled, but I can't find a way to listen to this type of event. In verbose logs, there's nothing written from https://github.com/koreader/koreader-base/blob/9e16a13a6dfe66dcb6ac3ed0b4b57b4355c01f9e/ffi/input_pocketbook.lua#L162.

Maybe there's some kind of a callback or event listener we could use? Digging through https://github.com/pocketbook/SDK_6.3.0/blob/dd8de463733a2bc0d53d1214b64734586fac427e/SDK-B288/usr/arm-obreey-linux-gnueabi/sysroot/usr/local/include/inkview.h was inefficient.

P.S. I'm aware I can toggle the frontlight by KOReader gesture, but don't want to stain the screen. OCD, I know.

**Comment by dmalinovsky** [comment]
In my naivete, I've contacted Pocketbook support for help. That's what they replied:

> We do not provide consultations on third-party applications that were installed on the device. If you have any questions regarding the use of PocketBook, feel free to contact us.

Seems like this itch remains unscratched.

**Reply by Frenzie** [reply]
I thought they had replied with technical info in the past (e.g., suggesting a particular function in inkview). Maybe the "wrong" email?

**Reply by Frenzie** [reply]
Pinging @rjd22

**Reply by rjd22** [reply, marked as answer]
Things might have changed but I used to contact them on help@pocketbook-int.com for technical questions. But it's really likely that if you're doing the front light change mapping with the pocketbook software no event will be shared with KoReader. Input events are mostly button, touch and gyro events. Changes like wifi or screen brightness don't really belong there.

**Reply by dmalinovsky** [reply]
The way I made it work is adding a gesture to toggle frontlight from KOReader itself, this way the status bar is aware of this and updates accordingly. But now the problem is that there's notification "Frontlight enabled/disabled" and it doesn't hide after a timeout. If I tap on it, the top menu opens. :)

So I just disabled gesture notifications altogether.

**Reply by NiLuJe** [reply]
The Notification issue stems from autostandby: #9493

**Reply by dmalinovsky** [reply]
Okay, thank you, @NiLuJe. I'll keep it disabled then.

**Reply by dmalinovsky** [reply]
Actually, I enabled Auto Standby plugin and notifications work perfectly! Seems like my habit of keeping as few  plugins running as possible can be troublesome. 
Thank you, @NiLuJe!
